apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -979,12 +986,15 @@ export class InstallManager {
           } catch (error) {
             // All the install attempts failed.
 
-            if (this._rushConfiguration.packageManager === 'pnpm') {
+            if (
+              this._rushConfiguration.packageManager === 'pnpm' &&
+              this._rushConfiguration.pnpmOptions.pnpmStore === 'local'
+            ) {
               // If the installation has failed even after the retries, then pnpm store may
               // have got into a corrupted, irrecoverable state. Delete the store so that a
               // future install can create the store afresh.
               console.log(colors.yellow(`Deleting the "pnpm-store" folder`));
-              this._commonTempFolderRecycler.moveFolder(this._rushConfiguration.pnpmStoreFolder);
+              this._commonTempFolderRecycler.moveFolder(this._rushConfiguration.pnpmOptions.pnpmStorePath);
```

Comment:
`rush purge --unsafe` should probably delete the global store. Is that case handled?

Reply:
Seems strange to me, considering I would consider that PNPM territory, not Rush; but I'm not opposed.  I can see how it would make sense under certain circumstances, and it is, after all "unsafe" :laughing:.

There's still extra work I need to do for determining the global store location in as many cases as possible (i.e. if it is user configured by pnpm configs, or if it is changed in `.npmrc`), in order to address https://github.com/microsoft/rushstack/pull/1657#issuecomment-594824405

Please let me know if you see anything erroneous in the `PurgeManager.purgeUnsafe()` implementation.  It's not 100% ready to go, yet.  I need to do some testing, but I've run out of time tonight.